core/runtime/v2/shim_load.go

```diff
@@ -71,48 +73,52 @@ func (m *ShimManager) loadShims(ctx context.Context, stateDir string) error {
 	if err != nil {
 		return err
 	}
+	eg, ctx2 := errgroup.WithContext(ctx)
+	eg.SetLimit(runtime.GOMAXPROCS(0))
 	for _, sd := range shimDirs {
-		if !sd.IsDir() {
-			continue
-		}
-		id := sd.Name()
-		// skip hidden directories
-		if len(id) > 0 && id[0] == '.' {
-			continue
-		}
-		bundle, err := LoadBundle(ctx, stateDir, id)
-		if err != nil {
-			// fine to return error here, it is a programmer error if the context
-			// does not have a namespace
-			return err
-		}
-		// fast path
-		f, err := os.Open(bundle.Path)
-		if err != nil {
-			bundle.Delete()
-			log.G(ctx).WithError(err).Errorf("fast path read bundle path for %s", bundle.Path)
-			continue
-		}
-
-		bf, err := f.Readdirnames(-1)
-		f.Close()
-		if err != nil {
-			bundle.Delete()
-			log.G(ctx).WithError(err).Errorf("fast path read bundle path for %s", bundle.Path)
-			continue
-		}
-		if len(bf) == 0 {
-			bundle.Delete()
-			continue
-		}
-		if err := m.loadShim(ctx, bundle); err != nil {
-			log.G(ctx).WithError(err).Errorf("failed to load shim %s", bundle.Path)
-			bundle.Delete()
-			continue
-		}
+		eg.Go(func() error {
+			if !sd.IsDir() {
+				return nil
+			}
+			id := sd.Name()
+			// skip hidden directories
+			if len(id) > 0 && id[0] == '.' {
+				return nil
+			}
```

Comment:
nit: can we move this out of goroutine and just continue to the next for loop? I feel it's not worthwhile to create goroutines for those cases.

Reply:
I think these code will not generate much  overhead  in go routine.  I think the main overhead is 
```
net.DialTimeout("unix", socket(address).path(), timeout)
```
to cnonect shim

Reply:
I mean we don't need to create a separate goroutines (by moving them out of the goroutine logic, so we can `continue` to the next forloop directly) in those cases

Reply:
done, thanks. @djdongjin